syntax = "proto3";
package storyboard;




import "next_builder_sdk/model/next_builder/import_storyboard_node.proto";
import "next_builder_sdk/model/next_builder/storyboard_brick.proto";
import "next_builder_sdk/model/next_builder/storyboard_route.proto";
import "next_builder_sdk/model/next_builder/micro_app_project.proto";

/*
ImportStoryboardNode请求
*/
message ImportStoryboardNodeRequest {
    /*
    storyboard node
    */
    next_builder.ImportStoryboardNode node = 1 ;
}

/*
ImportStoryboardNode返回
*/
message ImportStoryboardNodeResponse {
    /*
    实例标识
    */
    string instanceId = 1 ;
    /*
    别名
    */
    string alias = 2 ;
    /*
    micro_app ID
    */
    string appId = 3 ;
    /*
    节点标识
    */
    string id = 4 ;
    /*
    挂载点
    */
    string mountPoint = 5 ;
    /*
    排序
    */
    int32 sort = 6 ;
    /*
    类型
    */
    string type = 7 ;
    /*
    当type为brick时，填充该字段属性
    */
    next_builder.StoryboardBrick brick = 8 ;
    /*
    当type为route时，填充该字段属性
    */
    next_builder.StoryboardRoute route = 9 ;
    /*
    project
    */
    next_builder.MicroAppProject project = 10 ;
}

/*
ImportStoryboardNodeApi返回
*/
message ImportStoryboardNodeResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    storyboard.ImportStoryboardNodeResponse data = 4 ;
}

